js/src/common/utils/ItemList.ts

```diff
@@ -81,51 +169,143 @@ export default class ItemList {
    * Remove an item from the list.
    */
   remove(key: string): this {
-    delete this.items[key];
+    delete this._items[key];
 
     return this;
   }
 
   /**
    * Merge another list's items into this one.
+   *
+   * The list passed to this function will overwrite items which already exist
+   * with the same key.
    */
-  merge(items: this): this {
-    for (const i in items.items) {
-      if (items.items.hasOwnProperty(i) && items.items[i] instanceof Item) {
-        this.items[i] = items.items[i];
+  merge<K>(otherList: ItemList<K>): ItemList<T | K> {
+    const otherItems = otherList.toObject();
+
+    Object.keys(otherItems).forEach((key) => {
+      const val = otherItems[key];
+
+      if (val instanceof Item) {
+        (this as ItemList<T | K>)._items[key] = val;
```

Comment:
Doesn't this lose priority though?

Reply:
No, it's copying the whole Item class from the other ItemList and adding it to this one.